src/System.Net.Mime/src/System/Net/Mime/QEncodedStream.cs

```diff
@@ -0,0 +1,400 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using System.Text;
+using System.Net.Http;
+using System.Diagnostics;
+
+namespace System.Net.Mime
+{
+    /// <summary>
+    /// This stream performs in-place decoding of quoted-printable
+    /// encoded streams.  Encoding requires copying into a separate
+    /// buffer as the data being encoded will most likely grow.
+    /// Encoding and decoding is done transparently to the caller.
+    /// </summary>
+    internal class QEncodedStream : DelegatingStream, IEncodableStream
+    {
+        //folding takes up 3 characters "\r\n "
+        private const int SizeOfFoldingCRLF = 3;
+
+        private static readonly byte[] s_hexDecodeMap = new byte[]
+        {
+            // 0   1   2   3   4   5   6   7   8   9   A   B   C   D   E   F
+             255,255,255,255,255,255,255,255,255,255,255,255,255,255,255,255, // 0
+             255,255,255,255,255,255,255,255,255,255,255,255,255,255,255,255, // 1
+             255,255,255,255,255,255,255,255,255,255,255,255,255,255,255,255, // 2
+             0,  1,  2,  3,  4,  5,  6,  7,  8,  9,255,255,255,255,255,255, // 3
+             255, 10, 11, 12, 13, 14, 15,255,255,255,255,255,255,255,255,255, // 4
+             255,255,255,255,255,255,255,255,255,255,255,255,255,255,255,255, // 5
+             255, 10, 11, 12, 13, 14, 15,255,255,255,255,255,255,255,255,255, // 6
+             255,255,255,255,255,255,255,255,255,255,255,255,255,255,255,255, // 7
+             255,255,255,255,255,255,255,255,255,255,255,255,255,255,255,255, // 8
+             255,255,255,255,255,255,255,255,255,255,255,255,255,255,255,255, // 9
+             255,255,255,255,255,255,255,255,255,255,255,255,255,255,255,255, // A
+             255,255,255,255,255,255,255,255,255,255,255,255,255,255,255,255, // B
+             255,255,255,255,255,255,255,255,255,255,255,255,255,255,255,255, // C
+             255,255,255,255,255,255,255,255,255,255,255,255,255,255,255,255, // D
+             255,255,255,255,255,255,255,255,255,255,255,255,255,255,255,255, // E
+             255,255,255,255,255,255,255,255,255,255,255,255,255,255,255,255, // F
+        };
+
+        //bytes that correspond to the hex char representations in ASCII (0-9, A-F)
+        private static readonly byte[] s_hexEncodeMap = new byte[] { 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 65, 66, 67, 68, 69, 70 };
+
+        private ReadStateInfo _readState;
+        private WriteStateInfoBase _writeState;
+
+        internal QEncodedStream(WriteStateInfoBase wsi) :
+            base(new MemoryStream()) // TODO: what should this be?!?!!
```

Comment:
Thanks for catching this.  I'm going to leave a comment but change the text next to this.  The desktop code is essentially passing null as the inner stream, which suggests to me that this is dead code, as any usage of any member would then blow up with a null reference exception.  But since this code isn't used in this PR, it's hard for me to know for sure.